crates/wasmtime/src/trap.rs

```diff
@@ -226,4 +221,37 @@ impl fmt::Display for Trap {
     }
 }
 
-impl std::error::Error for Trap {}
+impl std::error::Error for Trap {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match &self.inner.reason {
+            TrapReason::Error(e) => Some(e.as_ref()),
```

Comment:
Since the error itself is printed as part of `Display` I think this will want to call `e.source()` since the `Trap` already represents the top-level error.

Reply:
Makes sense, did that.